### PR TITLE
Add 'remove_source_branch' option when creating an MR from a Github PR

### PR DESCRIPTION
**Pull request by Dazza0**
The `github_pr_to_internal_pr.py` script doesn't set the `remove_source_branch` option when creating the Gitlab MR. Developers often forget to set that option manually as well when reviewing the automatically created MR. Thus, when the MR is merged, we often get a lot of stale branches that still exist.

### CODE REVIEW

**Comment by Dazza0** [comment]
@laukik-hase PTAL

**Review by igrr — APPROVED** [review]
LGTM!

Another nice thing to have would be automatically assigning the user who triggered the action to the created MR. Otherwise it is sometimes left unassigned and abandoned. But i guess that would require keeping the user name mapping somewhere accessible for the action.

**Review by kumekay — APPROVED** [review]
LGTM, useful addition

**Review by igrr — APPROVED** [review]

**Comment by Dazza0** [comment]
@kumekay @igrr Any ideas on how I can test this before merging?

**Comment by laukik-hase** [comment]
@Dazza0 LGTM!

**Comment by laukik-hase** [comment]
> LGTM!
> 
> Another nice thing to have would be automatically assigning the user who triggered the action to the created MR. Otherwise it is sometimes left unassigned and abandoned. But i guess that would require keeping the user name mapping somewhere accessible for the action.


@igrr, we do have a mechanism of automatically assigning GitHub issues/PRs to developers when they are assigned in JIRA - could we make use of a similar mechanism in this case?

**Comment by mahavirj** [comment]
> @kumekay @igrr Any ideas on how I can test this before merging?

I can help you with that.

**Comment by mahavirj** [comment]
> > LGTM!
> > Another nice thing to have would be automatically assigning the user who triggered the action to the created MR. Otherwise it is sometimes left unassigned and abandoned. But i guess that would require keeping the user name mapping somewhere accessible for the action.
> 
> @igrr, we do have a mechanism of automatically assigning GitHub issues/PRs to developers when they are assigned in JIRA - could we make use of a similar mechanism in this case?

Alternatively, we can simply put a comment to notify all maintainers `@esp-idf-codeowners/all-maintainers`

**Comment by mahavirj** [comment]
Verified the functionality of the PR, we can add the notification related update in followup PR

**Pull request merged**
